src/main/java/com/redhat/devtools/intellij/quarkus/lsp4ij/LSPEditorWrapper.java

```diff
@@ -0,0 +1,246 @@
+                documentChanged = true;
+            }
+        };
+        this.editor.getDocument().addDocumentListener(documentListener);
+
+        this.caretListener = new CaretListener() {
+            @Override
+            public void caretPositionChanged(@NotNull CaretEvent event) {
+                if (documentChanged) {
+                    documentChanged = false;
+                    return;
+                }
+                int offset = editor.getCaretModel().getCurrentCaret().getOffset();
+                VirtualFile file = LSPIJUtils.getFile(editor.getDocument());
+                if (file != null) {
+                    updateQuickFixAt(offset, -1,-1);
+                }
+            }
+        };
+        this.editor.getCaretModel().addCaretListener(caretListener);
+    }
+
+    private static int getTargetOffset(EditorMouseEvent event) {
+        Editor editor = event.getEditor();
+        if (editor instanceof EditorEx &&
+                editor.getProject() != null &&
+                event.getArea() == EditorMouseEventArea.EDITING_AREA &&
+                event.getMouseEvent().getModifiers() == 0 &&
+                event.isOverText() &&
+                event.getCollapsedFoldRegion() == null) {
+            return event.getOffset();
+        }
+        return -1;
+    }
+
+    public Editor getEditor() {
+        return editor;
+    }
+
+    /**
+     * Update Intellij QuickFixes which should be available at the given offset
+     * by collecting LSP code actions at the given offset.
+     *
+     * @param offset the caret / hover offset
+     * @return
+     */
+    private CompletableFuture<Void> updateQuickFixAt(int offset, int start,int end) {
+        Document document = editor.getDocument();
+        VirtualFile file = LSPIJUtils.getFile(document);
+        if (file == null) {
+            return null;
+        }
+        URI fileUri = LSPIJUtils.toUri(document);
+
+        CodeActionParams params = new CodeActionParams();
+        params.setTextDocument(LSPIJUtils.toTextDocumentIdentifier(fileUri));
+        Position pos = LSPIJUtils.toPosition(offset, document);
+        Range range = new Range(pos, pos);
+        params.setRange(range);
+
+        List<Diagnostic> currentDiagnostics = LSPVirtualFileWrapper.getLSPVirtualFileWrapper(file).getAllDiagnostics();
+        List<Diagnostic> diagnosticContext = new ArrayList<>();
+        if (!currentDiagnostics.isEmpty()) {
+            List<Diagnostic> diagnostics = new ArrayList<>(currentDiagnostics);
+            diagnostics.forEach(diagnostic -> {
+                int startOffset = LSPIJUtils.toOffset(diagnostic.getRange().getStart(), document);
+                int endOffset = LSPIJUtils.toOffset(diagnostic.getRange().getEnd(), document);
+                if (offset >= startOffset && offset <= endOffset) {
+                    diagnosticContext.add(diagnostic);
+                }
+            });
+        }

```

Comment:
nitpicking: l146-156 could be separated into their own method ex. `createDiagnosticContext` and improve readability.